plugins/imfile/imfile.c

```diff
@@ -996,13 +1007,16 @@ act_obj_destroy(act_obj_t *const act, const int is_deleted)
 		pollFile(act); /* get any left-over data */
 		if(inst->bRMStateOnDel) {
 			statefn = getStateFileName(act, statefile, sizeof(statefile));
-			getFullStateFileName(statefn, "", toDel, sizeof(toDel)); // TODO: check!
+			if(!inst->inotifyNewBehaviour)
+				getFullStateFileName(statefn, "", toDel, sizeof(toDel)); // TODO: check!
+			else
+				getFullStateFileName(statefn, act->file_id, toDel, sizeof(toDel)); // TODO: check!
```

Comment:
I think we don't need to distinguish between old and new style anymore, due to https://github.com/rsyslog/rsyslog/pull/4633/commits/fca8e14d65019ee1b814c4ed16ec7ff04de30809#diff-cfff74f7d8ebca9eeb47097882fc54089b1973aeb56c4b2d8240f40f77b2ca38R999 but I might be wrong in this case.